crates/iroha_executor/src/default/isi/multisig/transaction.rs

```diff
@@ -0,0 +1,261 @@
+//! Validation and execution logic of instructions for multisig transactions
+
+use alloc::collections::{btree_map::BTreeMap, btree_set::BTreeSet};
+
+use super::*;
+
+impl VisitExecute for MultisigPropose {
+    fn visit<V: Execute + Visit + ?Sized>(&self, executor: &mut V) {
```

Comment:
this function is unnecessary as everything validated here will be re-validated upon execution

Reply:
- `is_downward_proposal` ... No, I don't find any execution parts near to checking `proposer` signatories or `target_account` roles
- `has_multisig_role` ... Yeah, checking if `target_account` signatories contains `proposer` during execution will be an equivalent
- `"multisig proposal duplicates"` ... No, I think `SetKeyValue<Account>` would just replace and initialize the proposal without this validation

Reply:
> - `has_multisig_role` ... Yeah, checking if `target_account` signatories contains `proposer` during execution will be an equivalent

However, the role and the metadata cannot be exactly a substitution for each other: #5229
I'd like to leave it as is to double check for now